tests/integration/aof-multi-part.tcl

```diff
@@ -469,25 +464,66 @@ tags {"external:skip"} {
             set d2 [$client debug digest]
             assert {$d1 eq $d2}
         }
+
+        clean_aof_persistence $aof_dirpath
+        exec cp tests/assets/rdb-preamble.aof $aof_old_name_old_path
     }
 
-    clean_aof_persistence $aof_dirpath
+    # Tests17: Multi Part AOF can load data from old version redis (rdb preamble yes)
+    test {Multi Part AOF can load data from old version redis (rdb preamble yes)} {
+        start_server_aof [list dir $server_path] {
+            assert_equal 1 [is_alive $srv]
 
-    # Tests17: Multi Part AOF can continue the upgrade from the interrupted upgrade state
-    create_aof $aof_old_version_filepath {
-        append_to_aof [formatCommand set k1 v1]
-        append_to_aof [formatCommand set k2 v2]
-        append_to_aof [formatCommand set k3 v3]
-    }
+            set client [redis [dict get $srv host] [dict get $srv port] 0 $::tls]
+            wait_done_loading $client
 
-    create_aof_dir $aof_dirpath
+            assert_equal v1 [$client get k1]
+            assert_equal v2 [$client get k2]
+            assert_equal v3 [$client get k3]
```

Comment:
i didn't bother to download and look at the binary file, so i don't know what's in it.
i assume some of these keys are part of the rdb header, and some are part of the AOF tail?
is that true? maybe add a comment here.

Reply:
all k1 k2 and k3 in rdb header， do we need rdb header and aof tail?

Reply:
yes, i wanna make sure we can still handle such files (since we no longer generate them)

Reply:
OK